Backend/TimeTracker.Api/Controllers/TimerController.cs

```diff
@@ -0,0 +1,206 @@
+
+            return new GenericResponseDTO<TimerDTO> {
+                Data = new TimerDTO {
+                    Id = timer.Id,
+                    StartTime = timer.StartTime,
+                    Notes = timer.Notes,
+                    ProjectId = timer.Project.Id
+                },
+                Success = true
+            };
+            
+        }
+
+        // returns a TimeEntryDTO for the newly created TimeEntry
+        [HttpPost]
+        [Route("Stop")]
+        public async Task<GenericResponseDTO<TimeEntryDTO>> StopTimer(int timerId) {
+
+            if(User == null) {
+                return new GenericResponseDTO<TimeEntryDTO> {
+                    Message = "Not logged in.",
+                    Success = false
+                };
+            }
+
+            Timer timer = await database.Timers
+                .Include(t => t.User)
+                .Include(t => t.Project)
+                .FirstOrDefaultAsync(t => t.Id == timerId);
+
+            if(timer.User.Id != authHelper.GetCurrentUserId(User)) {
+                return new GenericResponseDTO<TimeEntryDTO> {
+                    Message = "Invalid user.",
+                    Success = false
+                };
+            }
+
+            DateTime timeEntryCreationTime = DateTime.Now;
+
+            TimeEntry timeEntry = (await database.TimeEntries
+                .AddAsync(new TimeEntry{
+                    CreatedTime = timeEntryCreationTime,
+                    Notes = timer.Notes,
+                    Project = timer.Project,
+                    LastModified = timeEntryCreationTime,
+                    User = await authHelper.GetCurrentUser(User, database),
+                    Length = (timeEntryCreationTime - timer.StartTime).TotalMinutes,
+                    Day = timeEntryCreationTime
+                })).Entity;
+
+            TimeEntryDTO returnData = new TimeEntryDTO {
+                Id = timeEntry.Id,
+                Length = timeEntry.Length,
+                Notes = timeEntry.Notes,
+                ProjectId = timer.Project.Id,
+                Day = timeEntry.Day
+            };
+
+            database.Timers.Remove(timer);
+
+            await database.SaveChangesAsync();
+
+            return new GenericResponseDTO<TimeEntryDTO> {
+                Success = true,
+                Data = returnData
+            };
+        }
+
+        // returns a TimerDTO for the given timer id
+        [HttpPost]
+        public async Task<GenericResponseDTO<TimerDTO>> GetTimerById(int timerId) {
+            
+            if(User == null) {
+                return new GenericResponseDTO<TimerDTO> {
+                    Message = "Not logged in.",
+                    Success = false
+                };
+            }
+
+            Timer timer = await database.Timers
+                .AsNoTracking()
+                .Include(t => t.User)
+                .Include(t => t.Project)
+                .FirstOrDefaultAsync(t => t.Id == timerId);
```

Comment:
Add the check for the user ID here. Then you can also remove the `.Include(t => t.User)` call